src/library/scala/collection/immutable/Vector.scala

```diff
@@ -204,18 +204,23 @@ final class Vector[+A] private[immutable] (private[collection] val startIndex: I
     import Vector.{Log2ConcatFaster, TinyAppendFaster}
     if (prefix.isEmpty) this
     else {
-      prefix.size match {
-        case n if n <= TinyAppendFaster || n < (this.size >>> Log2ConcatFaster) =>
-          var v: Vector[B] = this
-          val it = prefix.toIndexedSeq.reverseIterator
-          while (it.hasNext) v = it.next() +: v
-          v
-        case n if this.size < (n >>> Log2ConcatFaster) && prefix.isInstanceOf[Vector[_]] =>
-          var v = prefix.asInstanceOf[Vector[B]]
-          val it = this.iterator
-          while (it.hasNext) v = v :+ it.next()
-          v
-        case _ => super.prependedAll(prefix)
+      prefix match {
+        case prefix: collection.Iterable[B] =>
+          prefix.size match {
```

Comment:
Should we be calling `knownSize` here?

Reply:
I agree. But this pull request only deal with the bug.

Reply:
👍  another PR is welcome!